Summary.java

```diff
@@ -42,68 +42,61 @@ public static void main(String[] args) throws IOException{
                         ? str.length() : sentences.get(i + 1);
-            String singleSentence = str.substring(begin, end);
-            
-            // TODO: there's a better way to do this right?
-            ArrayList<String> currentSentence = new ArrayList<String>();
-                
-            // remove punctuation and convert to lowercase and then split
-            // on white spaces
-            words = singleSentence.replaceAll("[^a-zA-Z ]", "")
-                    .toLowerCase().split("\\s+");
-            for(String singleWord: words){
-                if(occurance.containsKey(singleWord)){
-                    int count = occurance.get(singleWord) + 1;
-                    occurance.put(singleWord, count);
-                } else {
-                    occurance.put(singleWord, 1);
-                }
-                
-                if (!currentSentence.contains(singleWord)) {
-                    currentSentence.add(singleWord);
-                }
-            }
-            
-            for (String word : currentSentence) {
+
+            Sentence currSentence = new Sentence(str.substring(begin, end));
+            averageLength += currSentence.getLength();
+
+            for (String word : currSentence.getCurrentSentence()) {
                 if (sentenceCount.containsKey(word)) {
                     int count = sentenceCount.get(word) + 1;
                     sentenceCount.put(word, count);
                 } else {
                     sentenceCount.put(word, 1);
                 }
             }
+            theSentences.add(currSentence);
         }
-        
-        List<Map.Entry<String, Integer>> list = new ArrayList<>(occurance.entrySet());
-        list.sort(Map.Entry.comparingByValue());
+        averageLength /= (double) theSentences.size();
+
 
-        for(Map.Entry<String, Integer> entry : list){
-            System.out.printf("key: %s; occurences: %d; sentences: %d\n",
-                entry.getKey(), entry.getValue(),
-                sentenceCount.get(entry.getKey()));
-        }
 
-        System.out.printf("Number of sentences: %d\n", numSentences);
+        // List<Map.Entry<String, Integer>> list = new ArrayList<>(occurance.entrySet());
+        // list.sort(Map.Entry.comparingByValue());
+
+        // for(Map.Entry<String, Integer> entry : list){
+        //     System.out.printf("key: %s; occurences: %d; sentences: %d\n",
+        //         entry.getKey(), entry.getValue(),
+        //         sentenceCount.get(entry.getKey()));
+        // }
+        // System.out.printf("Number of sentences: %d\n", numSentences);
+
         /*
         TODO - replace everything up there with the sentence class and test it
         */
         
         /*
-        TODO - call the calculateBM25() function to get sentence rank
+        Call the calculateBM25() function to get sentence rank
         */
-        
+        //BM-25 Array
+        ArrayList<Double> bm25Array = new ArrayList<Double>();
```

Comment:
Change this `bm25Array` to a HashMap with sentence index as key and the score as the value.
Current set up has no way of matching the score with the corresponding sentence.